### PR TITLE
tests: add test for sock_cgroup_ptr function

### PR DESCRIPTION
**Pull request by marxin**
Related to: #253.

Signed-off-by: Martin Liska <mliska@suse.cz>

### CODE REVIEW

File: tests/linux_kernel/helpers/test_cgroup.py [hunk 1]
```diff
@@ -112,6 +120,14 @@ def test_cgroup_get_from_path(self):
             NULL(self.prog, "struct cgroup *"),
         )
 
+    def test_cgroup_socket(self):
+        with create_socket() as sock:
+            task = find_task(self.prog, os.getpid())
+            file = fget(task, sock.fileno())
+            sk = cast("struct socket *", file.private_data).sk
+            self.assertTrue(sk_fullsock(sk))
```
Comment (osandov) [inline tests/linux_kernel/helpers/test_cgroup.py:128]: There's already a test case for `sk_fullsock()`, so no need for this here.
```suggestion
```

File: tests/linux_kernel/helpers/test_cgroup.py [hunk 2]
```diff
@@ -112,6 +120,14 @@ def test_cgroup_get_from_path(self):
...
+            self.assertIsNotNone(sock_cgroup_ptr(sk.sk_cgrp_data))
```
Comment (osandov) [inline tests/linux_kernel/helpers/test_cgroup.py:129]: I think we can do even better and make sure that we get the correct cgroup:

```suggestion
            self.assertEqual(sock_cgroup_ptr(sk.sk_cgrp_data), task.cgroups.dfl_cgrp)
```

**Review by osandov — CHANGES_REQUESTED** [review]
Nice, thanks for adding this! A couple of minor comments.

**Comment by osandov** [comment]
It also looks like this needs to be rebased on the latest  `main` to get your `sock_cgroup_ptr` fix.

**Comment by marxin** [comment]
Ok, so it seems both approaches of `sock_cgroup_ptr` do fail on `4.9`:

```
2023-01-28T08:31:30.1702240Z ======================================================================
2023-01-28T08:31:30.1708842Z ERROR: test_cgroup_socket (tests.linux_kernel.helpers.test_cgroup.TestCgroup)
2023-01-28T08:31:30.1715524Z ----------------------------------------------------------------------
2023-01-28T08:31:30.1719255Z Traceback (most recent call last):
2023-01-28T08:31:30.1743498Z   File "/home/runner/work/drgn/drgn/drgn/helpers/linux/cgroup.py", line 45, in sock_cgroup_ptr
2023-01-28T08:31:30.1743911Z     return skcd.cgroup
2023-01-28T08:31:30.1744394Z AttributeError: 'struct sock_cgroup_data' has no member 'cgroup'
2023-01-28T08:31:30.1746975Z 
2023-01-28T08:31:30.1753115Z During handling of the above exception, another exception occurred:
2023-01-28T08:31:30.1753607Z 
2023-01-28T08:31:30.1755722Z Traceback (most recent call last):
2023-01-28T08:31:30.1758250Z   File "/home/runner/work/drgn/drgn/tests/linux_kernel/helpers/test_cgroup.py", line 127, in test_cgroup_socket
2023-01-28T08:31:30.1760682Z     self.assertEqual(sock_cgroup_ptr(sk.sk_cgrp_data), task.cgroups.dfl_cgrp)
2023-01-28T08:31:30.1763114Z   File "/home/runner/work/drgn/drgn/drgn/helpers/linux/cgroup.py", line 47, in sock_cgroup_ptr
2023-01-28T08:31:30.1763896Z     return cast("struct cgroup *", skcd.val)
2023-01-28T08:31:30.1764578Z AttributeError: 'struct sock_cgroup_data' has no member 'val'
```

What to do with that?

**Review by osandov — APPROVED** [review]
Thanks! The test itself looks good. It's failing on 4.9 because our kernel config doesn't select `CONFIG_SOCK_CGROUP_DATA` on 4.9. I just pushed f0beba13e25b77cb7560d8f2d494dc95f550e753 to enable it and am rebuilding the kernels now. It will take several hours. Once that's done, I'll rerun the tests to double check.

**Comment by marxin** [comment]
Great, thanks for the configuration change and new kernel images.

**Pull request merged**